KestrelMock/Services/MockService.cs

```diff
@@ -1,375 +1,75 @@
-﻿using Microsoft.AspNetCore.Http;
+﻿using KestrelMock.Settings;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using System;
-using System.Collections.Concurrent;
-using System.Collections.Generic;
 using System.IO;
 using System.Linq;
 using System.Net;
-using System.Reflection.Metadata;
-using System.Runtime.InteropServices.ComTypes;
-using System.Text;
-using System.Text.Json;
-using System.Text.RegularExpressions;
 using System.Threading.Tasks;
 
 namespace KestrelMock.Services
 {
     public class MockService
     {
-        private ConcurrentDictionary<string, HttpMockSetting> _pathMappings;
-        private ConcurrentDictionary<string, HttpMockSetting> _pathStartsWithMappings;
-        private ConcurrentDictionary<string, List<HttpMockSetting>> _bodyCheckMappings;
-        private ConcurrentDictionary<Regex, HttpMockSetting> _pathMatchesRegex;
-
-        private static readonly Regex UriTemplateParameterParser = new Regex(@"\{(?<parameter>[^{}?]*)\}", RegexOptions.Compiled);
-
         private readonly MockConfiguration _mockConfiguration;
         private readonly RequestDelegate _next;
 
         public MockService(IOptions<MockConfiguration> options, RequestDelegate next)
         {
-            _pathMappings = new ConcurrentDictionary<string, HttpMockSetting>();
-            _pathStartsWithMappings = new ConcurrentDictionary<string, HttpMockSetting>();
-            _bodyCheckMappings = new ConcurrentDictionary<string, List<HttpMockSetting>>();
-            _pathMatchesRegex = new ConcurrentDictionary<Regex, HttpMockSetting>();
             _mockConfiguration = options.Value;
             _next = next;
         }
 
         public async Task Invoke(HttpContext context)
         {
-            if (_pathMappings.IsEmpty && _pathStartsWithMappings.IsEmpty && _bodyCheckMappings.IsEmpty)
-            {
-                SetupPathMappings(_mockConfiguration);
-                await LoadBodyFromFile();
-            }
+            var mappings = await InputMappingParser.ParsePathMappings(_mockConfiguration);
```

Comment:
Do we want to load these each time? I worry about the reading from disk each request slowing things down or is mappings somehow only loaded once?